### PR TITLE
Distroless version of proxy_init image

### PR DESCRIPTION
**Pull request by c0d1ngm0nk3y**

### CODE REVIEW

File: pilot/docker/Dockerfile.proxy_init [hunk 1]
```diff
@@ -1,8 +1,24 @@
-FROM ubuntu:xenial
+FROM ubuntu:xenial as builder
 RUN apt-get update && apt-get upgrade -y && apt-get install -y \
     iproute2 \
     iptables \
  && rm -rf /var/lib/apt/lists/*
 
+FROM gcr.io/distroless/base:debug as distroless
+
```
Comment (sdake) [inline pilot/docker/Dockerfile.proxy_init]: why two `FROM`?  I understand the builder part, and that seems finee, but not sure on this second and third from.  Could you expand?

File: pilot/docker/Dockerfile.proxy_init [hunk 2]
```diff
@@ -1,8 +1,24 @@
...
+COPY --from=builder /lib/xtables /lib/xtables
+COPY --from=builder /lib/x86_64-linux-gnu /lib/x86_64-linux-gnu
+COPY --from=builder /sbin/iptables* /sbin/
+COPY --from=builder /sbin/ip6tables* /sbin/
+
+COPY --from=builder /etc/iproute2 /etc/iproute2
+COPY --from=builder /bin/ip /bin/ip
+COPY --from=builder /sbin/ip /sbin/ip
+
+COPY --from=distroless  /busybox/sh /bin/sh
```
Comment (sdake) [inline pilot/docker/Dockerfile.proxy_init]: is copying bash from the builder image not possible here?  The busybox developer is very litigious...

Cheers
-steve

File: pilot/docker/Dockerfile.proxy_init [hunk 3]
```diff
@@ -1,8 +1,24 @@
-FROM ubuntu:xenial
+FROM ubuntu:xenial as builder
 RUN apt-get update && apt-get upgrade -y && apt-get install -y \
     iproute2 \
     iptables \
  && rm -rf /var/lib/apt/lists/*
 
+FROM gcr.io/distroless/base:debug as distroless
+
+FROM gcr.io/distroless/base:debug
```
Comment (sdake) [inline pilot/docker/Dockerfile.proxy_init]: will this work with just base (sans :debug) tag?

**Comment by sdake** [comment]
> /test istio-pilot-multicluster-e2e

this test cae has been broken forever ;)

**Comment by sdake** [comment]
@costinm - I think this image is probably the most complex of those wanting distroless.  I will schedule some time in the next env WG meeting related to this work.  It looks pretty good to me.   I have a bit of curiosity if it makes sense to ship two different sets of binaries to dockerhub - distroless, and non-distroless,.

Cheers
-steve

**Comment by c0d1ngm0nk3y** [comment]
> this test cae has been broken forever ;)

Normally, I see 2 options in such cases

a) Fix it
b) Throw it out

I think {{Project Mauve}} will address a) :)

**Comment by jsPlachetka** [comment]
Got merged into #13579.

**Pull request closed without merging**